### PR TITLE
add documentation about test structure and stub command

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
as title say. only doc :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +91,11 @@ describe "my server"
 end
 ```
 
+### Stubbing
+
```
Comment (rylnd) [inline README.md]: Please delete this inconsistent newline.

File: README.md [hunk 2]
```diff
@@ -82,6 +91,11 @@ describe "my server"
...
+You can stub commands using `stub_command`. This function takes as first argument the name of the command you will stub. You can provide an optional argument that will be the body of the method. (code that would be evaluated)
```
Comment (rylnd) [inline README.md]: "This function takes the name of the command you wish to stub. If provided, the second argument will be used as the body of the command."

You should definitely link to the example tests [as originally discussed](https://github.com/rylnd/shpec/pull/49#discussion_r25998988).

**Comment by rylnd** [comment]
In general, I feel that the tests themselves should (and do) serve as documentation, and I've already tried to point readers of the README to them as much as possible. If the tests fail to document all of the aspects of a given feature, we should add more of them :wink:.

File: README.md [hunk 3]
```diff
@@ -26,6 +26,15 @@ If you'd like your tests to run automatically when they change, we recommend the
 ```bash
 find . -name "*_shpec.sh" | entr shpec
 ```
+### Test structure
```
Comment (rylnd) [inline README.md]: Maybe this should be "Test Syntax" ?

File: README.md [hunk 4]
```diff
@@ -26,6 +26,15 @@ If you'd like your tests to run automatically when they change, we recommend the
...
+`Shpec` would be very familiar if you already know *BDD*, and test framework such as `rspec`, `jasmine`, `mocha`, *et caetera*
```
Comment (rylnd) [inline README.md]: `shpec`'s syntax is very similar to other BDD frameworks such as `RSpec`, `jasmine`, or `mocha`. (add github links, please).

File: README.md [hunk 5]
```diff
@@ -26,6 +26,15 @@ If you'd like your tests to run automatically when they change, we recommend the
...
+
+The two main constructs are `describe/end` to regroup test, and `it/end` to describe a specific test.
```
Comment (rylnd) [inline README.md]: What do you mean by "regroup?"

**Comment by locochris** [comment]
@rylnd - perhaps an update of the screenshot would also be good?

File: README.md [hunk 6]
```diff
@@ -26,6 +26,15 @@ If you'd like your tests to run automatically when they change, we recommend the
...
+
+In the `it` block you will be the assertions, whose syntax is describe in the next chapter.
+
+Since you spec will be sourced as any shell script you can put in it any shell command. (in fact `describe`, `it`, `end`, *matchers* are defined has shell function)
```
Comment (rylnd) [inline README.md]: This is a good thing to note! Let me think if there's a better way to word this...

File: README.md [hunk 7]
```diff
@@ -26,6 +26,15 @@ If you'd like your tests to run automatically when they change, we recommend the
 ```bash
 find . -name "*_shpec.sh" | entr shpec
 ```
+### Test structure
+`Shpec` would be very familiar if you already know *BDD*, and test framework such as `rspec`, `jasmine`, `mocha`, *et caetera*
+
+The two main constructs are `describe/end` to regroup test, and `it/end` to describe a specific test.
+
+In the `it` block you will be the assertions, whose syntax is describe in the next chapter.
```
Comment (rylnd) [inline README.md]: Where's the next chapter? :stuck_out_tongue_winking_eye:

File: README.md (hunk 5 above)
Comment (AdrieanKhisbe) [inline README.md]: I meant grouping in fact. (francism due to innatention)

File: README.md (hunk 3 above)
Comment (AdrieanKhisbe) [inline README.md]: Mmmmmm, don't know. Sure *test structure* doesn't sound that well. But I'm not that convinced either by *test syntax*

File: README.md (hunk 4 above)
Comment (AdrieanKhisbe) [inline README.md]: willdo

File: README.md (hunk 6 above)
Comment (AdrieanKhisbe) [inline README.md]: (about what part of the hunk are you referring?)

File: README.md (hunk 7 above)
Comment (AdrieanKhisbe) [inline README.md]: Coooome on.. :p I meant section haha?

File: README.md (hunk 3 above)
Comment (AdrieanKhisbe) [inline README.md]: What abut : *How to structure your Specs* (or test)

File: README.md (hunk 2 above)
Comment (AdrieanKhisbe) [inline README.md]: ooops, had forgot about it. done

**Comment by AdrieanKhisbe** [comment]
I handled mst of the comments.

Naming of *Tests Structure* section. is still subject to suggestions.

Here are some things to consider before merging: 

- adding an example block in test structure. `it titi toto; end...`
- adding some text to the matcher section, and maybe rename it Assertions
- some other ting you might point at :)

**Comment by AdrieanKhisbe** [comment]
@rylnd any reco, comment?

Otherwise we sould merge :)

File: README.md [hunk 8]
```diff
@@ -82,6 +91,14 @@ describe "my server"
 end
 ```
 
+### Stubbing
+You can stub commands using `stub_command`.
+This function takes the name of the command you wish to stub. If provided, the second argument will be used as the body of the command. (code that would be evaluated)
+Once you're done, you can delete it with `unstub_command`.
+
+The best example is the [shpec test for these feature](https://github.com/rylnd/shpec/blob/master/shpec/shpec_shpec.sh#L72-L89).
```
Comment (rylnd) [inline README.md]: Should be "shpec test for this feature"

File: README.md [hunk 9]
```diff
@@ -26,13 +26,22 @@ If you'd like your tests to run automatically when they change, we recommend the
 ```bash
 find . -name "*_shpec.sh" | entr shpec
 ```
+### Tests Structure
```
Comment (rylnd) [inline README.md]: How about 'Structuring your Tests'?

File: README.md [hunk 10]
```diff
@@ -26,13 +26,22 @@ If you'd like your tests to run automatically when they change, we recommend the
...
+`Shpec` would be very familiar if you already know *BDD*, and test frameworks such as [`rspec`](https://github.com/rspec/rspec), [`jasmine`](https://github.com/jasmine/jasmine), [`mocha`](https://github.com/mochajs/mocha), *et caetera*
```
Comment (rylnd) [inline README.md]: `shpec` is similar to other BDD frameworks like RSpec, Jasmine, or Mocha.

**Comment by AdrieanKhisbe** [comment]
fixed :)

**Comment by rylnd** [comment]
@AdrieanKhisbe I don't know why my other comments aren't showing up in the main log. Please see [this](https://github.com/rylnd/shpec/pull/75/files#r29703968) and [this](https://github.com/rylnd/shpec/pull/75/files#r29703920)

**Comment by AdrieanKhisbe** [comment]
*github*'s secrets...  :)

**Comment by rylnd** [comment]
:+1:

**Pull request merged**